docs(uwsgi): clarify uwsgi+gevent support (#7493)
DataDog/dd-trace-py: docs/advanced_usage.rst

- Threads must be enabled with the `enable-threads <https://uwsgi-docs.readthedocs.io/en/latest/Options.html#enable-threads>`__ or `threads <https://uwsgi-docs.readthedocs.io/en/latest/Options.html#threads>`__ options.
- Lazy apps must be enabled with the `lazy-apps <https://uwsgi-docs.readthedocs.io/en/latest/Options.html#lazy-apps>`__ option.
- For automatic instrumentation (like ``ddtrace-run``) set the `import <https://uwsgi-docs.readthedocs.io/en/latest/Options.html#import>`__ option to ``ddtrace.bootstrap.sitecustomize``.

> - Gevent monkey patching must be applied via `--gevent-monkey-patch <https://uwsgi-docs.readthedocs.io/en/latest/Gevent.html#monkey-patching>`. Using ``import gevent; gevent.patch_all(..)`` is not supported for uwsgi. 

Author: ```suggestion
- Gevent patching should NOT be enabled via `--gevent-patch <https://uwsgi-docs.readthedocs.io/en/latest/Gevent.html#monkey-patching>` option. Enabling gevent patching for the builtin threading library is NOT supported. Instead use ``import gevent; gevent.monkey.patch_all(thread=False)`` in your application. 
```
